app/src/Pages/Portal/Contributions/Utils/ContributionsFields.js

```diff
@@ -401,31 +459,67 @@ export const validate = values => {
     typeOfContributor,
   } = values;
   const error = {};
+  const visible = {};
 
-  if (paymentMethod === 'Check' && !checkNoEmptyString(checkNumber)) {
+  // Make these areas visible
+  visible.isPerson = !!(
+    typeOfContributor === ContributorTypeEnum.INDIVIDUAL ||
+    typeOfContributor === ContributorTypeEnum.FAMILY
+  );
+
+  visible.checkSelected = !!(
+    values.paymentMethod === 'Check' || values.paymentMethod === 'Money Order'
+  );
+
+  // Only show Employer section if the contributor type is Individual OR Family AND Occupation is 'Employed'
+  visible.emptyOccupationLetterDate = values.occupationLetterDate === '';
+  visible.showEmployerSection =
+    values.occupation === 'Employed' && visible.isPerson;
+  visible.showInKindFields = !!inKindContributionValues.includes(
+    values.subTypeOfContribution
+  );
+  // These fields are conditionally required
+  if (visible.checkSelected && !checkNoEmptyString(checkNumber)) {
     error.checkNumber = 'Check number is required.';
   }
-  if (paymentMethod === 'Money Order' && !checkNoEmptyString(checkNumber)) {
+  if (visible.checkSelected && !checkNoEmptyString(checkNumber)) {
     error.checkNumber = 'Money Order number is required.';
   }
```

Comment:
I think should be `visible.moneyOrder` on line 485

Reply:
Actually I see that it could either be "Check" or "Money Order", but don't see how the logic will get to the money order one.

Reply:
Fixed